velox/exec/GroupingSet.h

```diff
@@ -91,6 +91,18 @@ class GroupingSet {
     return table_ ? table_->numDistinct() : 0;
   }
 
+  /// Returns number of global grouping sets.
+  vector_size_t numGlobalGroupingSets() const {
+    return globalGroupingSets_.size();
+  }
+
+  /// If there are global grouping sets, then returns if they have default
+  /// output incase no input rows were received.
+  bool hasDefaultGlobalGroupingSetOutput() const {
+    return noMoreInput_ && numInputRows_ == 0 && !globalGroupingSets_.empty() &&
+        (isRawInput_ || isPartial_);
```

Comment:
> isRawInput_ || isPartial_

Can this be just isRawInput_? isPartial_ implies isRawInput_, no?

Reply:
@mbasmanova :
```
isPartial_ = step == core::AggregationNode::Step::kPartial ||
      step == core::AggregationNode::Step::kIntermediate;

```
and 

```
isRawInput_ = step == core::AggregationNode::Step::kPartial ||
      step == core::AggregationNode::Step::kSingle;
```

So I wrote is_RawInput_ || isPartial_.

The java code wrote it this way as well, so I followed that. But I don't think this code should get triggered for kIntermediate.

Reply:
@aditi-pandit I think this code should apply only to kSingle and kPartial, not to kIntermediate or kFinal. I guess the check for kIntermediate happens to work because intermediate never receives empty input.

Reply:
Have modified the code.